controllers/workspace/provision/pull_secret.go

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright (c) 2019-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package provision
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/devfile/devworkspace-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PullSecretsProvisioningStatus struct {
+	ProvisioningStatus
+	v1alpha1.PodAdditions
+}
+
+func PullSecrets(clusterAPI ClusterAPI) PullSecretsProvisioningStatus {
+	labelSelector, err := labels.Parse(fmt.Sprintf("%s=%s", config.DevWorkspacePullSecretLabel, "true"))
+	if err != nil {
+		return PullSecretsProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{
+				Err:     err,
+				Requeue: true,
```

Comment:
If we're returning `err != nil`, `Requeue: true` is not necessary, as returning an error always triggers a requeue.
```suggestion
```

Reply:
Note also that this condition (which we should never hit) is a failure state, so it should return `FailStartup: true`

Reply:
Should be fixed now, on parsing labels we fails startup, on error during list secrets, return error without requeue: true